docker/package/ubuntu.py

```diff
@@ -16,23 +16,26 @@ def build_ubuntu_package(
     source_archive_path: str = None,
     binaries_dir: str = None,
 ):
-    # ubuntu prohibits uppercase in packages names
-    pkg_name = pkg.name.lower()
-    old_version = pkg.meta.version
-    # debian build utils don't like '_' symbol in version
-    fixed_version = pkg.meta.version.replace("_", "-")
-    dir = f"{pkg_name}-{fixed_version}"
-    cwd = os.path.dirname(__file__)
-    date = subprocess.check_output(["date", "-R"]).decode().strip()
-    if source_archive_path is None:
-        pkg.fetch_sources(dir, binaries_dir)
-        pkg.gen_buildfile("/".join([dir, pkg.buildfile]), binaries_dir)
-        subprocess.run(["tar", "-czf", f"{dir}.tar.gz", dir], check=True)
-    else:
-        shutil.copy(f"{cwd}/../{source_archive_path}", f"{dir}.tar.gz")
-        subprocess.run(["tar", "-xzf", f"{dir}.tar.gz"], check=True)
-    pkg.meta.version = fixed_version
     for ubuntu_version in ubuntu_versions:
+        # ubuntu prohibits uppercase in packages names
```

Comment:
Why this change? weren't the definitions above working all the same and in common to different ubuntu versions? :thinking:

Reply:
It's because ubuntu quirk. `python3.8` is available on `bionic` and `focal`, but not on `jammy`. So almost everythere we need dispatching according to version. It could be solved if we use third party launchpads so we could supply `python3.8` as dep for every ubuntu version.

Reply:
I see, got it :ok_hand: 

> It could be solved if we use third party launchpads

Nope, I don't think that's worth the trouble